.github/workflows/npm.yml

```
@@ -0,0 +1,50 @@
+
+          SHOULD_PUBLISH=$(node -p "
+            const semver = require('semver');
+            semver.gt('$CURRENT_VERSION', '$PUBLISHED_VERSION') ? 'true' : 'false';
+          ")
+
+          echo "publish=$SHOULD_PUBLISH" >> "$GITHUB_OUTPUT"
+
+      - name: Publish to npm
+        if: steps.compare.outputs.publish == 'true'
+        run: npm ci
+        run: npm publish --provenance --access public
```

actionlint (1.7.4)

key "run" is duplicated in element of "steps" section. previously defined at line:47,col:9

(syntax-check)